ten-simple-rules-dockerfiles.Rmd

```diff
@@ -418,6 +442,9 @@ For example, software versions, maintainer contact information, along with vendo
 The OCI Image Format Specification provides some common label keys (see the "Annotations" section in @opencontainers_image-spec_2017) to help standardize field names across container tools, as shown below.
 These labels match the `org.label-schema.*` specification, which has been deprecated in favour or the new namespace but are still found a lot in existing containers.
 
+
+[SJE: this was new to me.  On the one hand, I like the idea of LABEL, but have they really been adopted by the community?  The document from 2017 cited above hasn't moved on since Jun 2017.  Is this really a thing?  And is the Dockerfile the right place for all this metadata?  It feels like a bit too fine-grained for my liking.  But if this is good practice, then I shall start adopting it!  ]
+
```

Comment:
LABEL directives are slow to be adopted, not adopted by most users for quick work, but if you look around they are used. The problem with labels isn't specific to labels, but reflects an overall systemic problem of people not documenting / setting metadata generally.

Reply:
Ok.  I'm guilty of not providing metadata, but part of doing this is to get people (including me) to adopt good habits, so I'll learn...